### PR TITLE
Landing pages UI

### PR DESCRIPTION
**Pull request by ngumokenneth**
https://3.basecamp.com/5023736/buckets/38393000/todos/7682282318/

### CODE REVIEW

File: lib/phx_tools_web/components/core_components.ex [hunk 1]
```diff
@@ -83,18 +83,18 @@ defmodule PhxToolsWeb.CoreComponents do
...
+        <Heroicons.information_circle :if={@kind == :info} mini class="w-4 h-4" />
+        <Heroicons.exclamation_circle :if={@kind == :error} mini class="w-4 h-4" />
         <%= @title %>
       </p>
       <p class="mt-2 text-[0.8125rem] leading-5"><%= msg %></p>
       <button
         :if={@close}
         type="button"
-        class="group absolute top-2 right-1 p-2"
+        class="absolute p-2 group top-2 right-1"
```
Comment (amos-kibet) [inline lib/phx_tools_web/components/core_components.ex:94]: Why this change?

File: lib/phx_tools_web/components/core_components.ex [hunk 2]
```diff
@@ -83,18 +83,18 @@ defmodule PhxToolsWeb.CoreComponents do
...
         aria-label={gettext("close")}
       >
-        <Heroicons.x_mark solid class="h-5 w-5 stroke-current opacity-40 group-hover:opacity-70" />
+        <Heroicons.x_mark solid class="w-5 h-5 stroke-current opacity-40 group-hover:opacity-70" />
```
Comment (amos-kibet) [inline lib/phx_tools_web/components/core_components.ex:97]: Why this change?

File: lib/phx_tools_web/components/core_components.ex [hunk 3]
```diff
@@ -118,7 +118,7 @@ defmodule PhxToolsWeb.CoreComponents do
         navigate={@navigate}
         class="text-sm font-semibold leading-6 text-zinc-900 hover:text-zinc-700"
       >
-        <Heroicons.arrow_left solid class="w-3 h-3 stroke-current inline" />
+        <Heroicons.arrow_left solid class="inline w-3 h-3 stroke-current" />
```
Comment (amos-kibet) [inline lib/phx_tools_web/components/core_components.ex:121]: Why this change?

**Review by amos-kibet — CHANGES_REQUESTED** [review]

**Review by amos-kibet — CHANGES_REQUESTED** [review]

File: lib/phx_tools_web/live/phx_tools_live/landing_component.ex [hunk 4]
```diff
@@ -3,38 +3,71 @@ defmodule PhxToolsWeb.PhxToolsLive.LandingComponent do
...
-            <div class="flex justify-end sm:py-4 md:py-0 md:order-last md:mx-2">
-              <img
-                src="/images/info_icon.svg"
-                alt="installation command info"
-                phx-click={JS.show(to: "#installation-instructions")}
+          <div class="md:flex lg:w-[75%] flex-col relative">
+            <div class="md:flex">
+              <div class="flex justify-end sm:py-3 md:order-last md:mx-2">
+                <img
+                  src="/images/info_icon.svg"
```
Comment (amos-kibet) [inline lib/phx_tools_web/live/phx_tools_live/landing_component.ex:30]: Use Phoenix verified routes.

File: lib/phx_tools_web/live/phx_tools_live/landing_component.ex [hunk 5]
```diff
@@ -3,38 +3,71 @@ defmodule PhxToolsWeb.PhxToolsLive.LandingComponent do
...
+                  alt="installation command info"
+                  phx-click={JS.show(to: "#installation-instructions")}
+                />
+              </div>
+              <div class="border sm:px-3 bg-[#26168780] w-full flex items-center justify-center py-2 sm:space-x-3 rounded-sm">
+                <h1 id="tool-installation" class="text-center text-white font-martian sm:text-sm">
+                  <%= render_slot(@installation_command) %>
+                </h1>
+                <div id="copy" phx-hook="CopyHook" class="sm:py-3 md:my-0">
+                  <img src="/images/clipboard.svg" alt="copy icon" />
```
Comment (amos-kibet) [inline lib/phx_tools_web/live/phx_tools_live/landing_component.ex:40]: Use Phoenix verified routes.

**Review by amos-kibet — CHANGES_REQUESTED** [review]

File: lib/phx_tools_web/components/core_components.ex [hunk 6]
```diff
@@ -34,25 +34,28 @@ defmodule PhxToolsWeb.CoreComponents do
...
+          <img src={@os_icon} class="h-7 w-7 " />
+          <h1 class="text-center text-white font-martian"><%= @os_name %></h1>
+        </div>
       </.link>
+      <div class="border-[2px] border-[#755FFF] rounded-xl cursor-pointer hover:bg-indigo-850 flex font-martian md:w-44">
+        <.link
+          navigate={@source_code}
+          class="flex items-center justify-center w-full space-x-2 text-sm rounded-xl"
+        >
+          <span><img src="/images/Icon.svg" alt="" /></span>
```
Comment (amos-kibet) [inline lib/phx_tools_web/components/core_components.ex:55]: Use Phoenix verified routes

File: lib/phx_tools_web/components/layouts/root.html.heex [hunk 7]
```diff
@@ -19,11 +19,12 @@
...
 
-  <body class="bg-[#110A33] mx-auto  sm:px-3 md:max-w-4xl sm:py-8">
+  <body class="bg-[#110A33] mx-auto  sm:px-2 md:px-7  lg:w-9/12 sm:py-8">
     <%= @inner_content %>
-    <div class="relative flex items-center justify-center lg:mb-[22px] md:mb-[33px] sm:mb-[35px] mt-[5%] px-[30%]">
-      <.link target="_blank" href="https://optimum.ba">
-        <SharedComponents.optimum_logo />
+    <div class="relative flex items-center justify-center lg:mb-[22px] md:mb-[33px] sm:mb-[35px] mt-[20%] px-[30%]">
+      <.link target="_blank" href="https://optimum.ba" class="flex items-center gap-3">
+        <img src="/images/optimum_logo.svg" alt="optimum logo" class="w-[36.45px] h-[36.45px]" />
```
Comment (amos-kibet) [inline lib/phx_tools_web/components/layouts/root.html.heex:27]: Use Phoenix verified routes.

File: lib/phx_tools_web/components/core_components.ex [hunk 8]
```diff
@@ -45,6 +46,14 @@ defmodule PhxToolsWeb.CoreComponents do
         <h1 class="text-center text-white font-martian"><%= @os_name %></h1>
       </div>
     </.link>
+    <div class="border-[2px] border-[#755FFF] rounded-xl cursor-pointer hover:bg-indigo-850 flex font-martian md:w-44">
+      <.link
+        navigate={@source_code}
+        class="flex items-center justify-center w-full space-x-2 text-sm rounded-xl"
+      >
+        <span><img src="/images/Icon.svg" alt="" /></span> <span>Source code</span>
```
[the remark this answers is not in the document]
Comment (ngumokenneth) [reply lib/phx_tools_web/components/core_components.ex]: Kindly expound on this

[the remark this answers is not in the document]
Comment (amos-kibet) [reply lib/phx_tools_web/components/core_components.ex]: For example, this code:
`<span></span> <span>Source code</span>`
should be rewritten as:
`<span></span> <span>Source code</span>`

File: lib/phx_tools_web/live/phx_tools_live/landing_component.ex [hunk 9]
```diff
@@ -3,44 +3,80 @@ defmodule PhxToolsWeb.PhxToolsLive.LandingComponent do
...
-              os_icon={~p"/images/linux.png"}
-              os_name="Linux"
-              class={[
-                "px-1",
-                @operating_system == "Linux" && "bg-indigo-850"
-              ]}
+      <div class="min-h-full">
+        <div class=" mt-[5%] md:flex flex-col items-center">
+          <div class="sm:flex flex-col items-center">
+            <img src="/images/Phx.tools.svg" alt="phx tools logo" class=" sm:w-4/12 " />
```
Comment (amos-kibet) [inline lib/phx_tools_web/live/phx_tools_live/landing_component.ex:21]:
```suggestion
            <img src={static_url(Endpoint, ~p"/images/Phx.tools.svg")} alt="phx tools logo" class=" sm:w-4/12 " />
```

File: assets/js/hooks/copy_hook.js [hunk 10]
```diff
@@ -1,12 +1,36 @@
 export default {
   mounted() {
+    const copySvg = `
```
Comment (almirsarajcic) [inline assets/js/hooks/copy_hook.js:3]: Let's not mix HTML and JS code. Put these two icons in the HEEX template and then toggle them (show one and hide the other one) once the command has been copied, and then revert after the timeout.

File: lib/phx_tools_web/components/core_components.ex [hunk 11]
```diff
@@ -83,8 +56,8 @@ defmodule PhxToolsWeb.CoreComponents do
       {@rest}
     >
       <p :if={@title} class="flex items-center gap-1.5 text-[0.8125rem] font-semibold leading-6">
-        <Heroicons.information_circle :if={@kind == :info} mini class="h-4 w-4" />
-        <Heroicons.exclamation_circle :if={@kind == :error} mini class="h-4 w-4" />
+        <Heroicons.information_circle :if={@kind == :info} mini class="w-4 h-4" />
+        <Heroicons.exclamation_circle :if={@kind == :error} mini class="w-4 h-4" />
```
Comment (almirsarajcic) [inline lib/phx_tools_web/components/core_components.ex:60]: Seems like an unnecessary change.

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+  use PhxToolsWeb, :html
+
+  alias Phoenix.LiveView.Utils
+  alias PhxToolsWeb.PhxToolsLive.Icons
+
+  @type assigns :: map()
+  @type rendered :: Phoenix.LiveView.Rendered.t()
+
+  @spec button(assigns()) :: rendered()
+  def button(%{live_action: :linux} = assigns) do
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_components.ex:13]: The name could be more comprehensive.

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+    ~H"""
+    <.os_link_card
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_components.ex:15]: There is some repetition going on in these two buttons.

File: lib/phx_tools_web/live/phx_tools_live/index.html.heex [hunk 14]
```diff
@@ -1,52 +1,22 @@
 <%= case @live_action do %>
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_live/index.html.heex:1]: Maybe switch to a component and pattern match in the function head instead.

File: lib/phx_tools_web/live/phx_tools_live/index.ex [hunk 15]
```diff
@@ -17,16 +17,29 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
...
+      "Mac" -> {:noreply, assign(socket, :live_action, :macOS)}
+      _other -> {:noreply, socket}
+    end
+  end
 
   defp apply_action(socket, :linux) do
-    assign(socket, seo_attributes: %{url: url(~p"/linux")})
+    {:noreply,
+     socket
+     |> assign(:live_action, :linux)
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_live/index.ex:35]: Isn't `live_action` already a part of the socket assigns?

File: lib/phx_tools_web/live/phx_tools_live/index.ex [hunk 16]
```diff
@@ -17,16 +17,29 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
...
+     |> assign(seo_attributes: %{url: url(~p"/linux")})}
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_live/index.ex:36]: Can this be provided dynamically instead?

File: lib/phx_tools_web/live/phx_tools_live/index.ex [hunk 17]
```diff
@@ -17,16 +17,29 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
...
   end
 
   defp apply_action(socket, :macOS) do
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_live/index.ex:39]: Can you combine these two so you don't have to hardcode assigning seo attributes, but depend on the live_action passed to it instead?

File: lib/phx_tools_web/live/phx_tools_live/unsupported_os_component.ex [hunk 18]
```diff
@@ -0,0 +1,60 @@
+defmodule PhxToolsWeb.PhxToolsLive.UnsupportedOsComponent do
+  use PhxToolsWeb, :html
+
+  alias PhxToolsWeb.PhxToolsComponents
+  alias PhxToolsWeb.PhxToolsLive.Icons
+
+  @spec unsupported_os_page(map()) :: Phoenix.LiveView.Rendered.t()
+  def unsupported_os_page(assigns) do
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_live/unsupported_os_component.ex:8]: Maybe put this together with the other OS components.

**Review by almirsarajcic — CHANGES_REQUESTED** [review]
Remove the unnecessary images.

File: assets/js/hooks/copy_hook.js [hunk 19]
```diff
@@ -1,12 +1,22 @@
...
-      copyBtn.innerHTML = 'Copied'
-      copyBtn.setAttribute('disabled', '')
-      copyBtn.classList.add('bg-[#9887FF]')
+      copyElement.classList.add('hidden')
+      copyConfirmationElement.classList.remove('hidden')
+
+      setTimeout(() => {
+        copyConfirmationElement.classList.add('hidden')
+        copyElement.classList.remove('hidden')
+      }, 400)
```
Comment (almirsarajcic) [inline assets/js/hooks/copy_hook.js:19]: Is this too fast? Have you checked with Oghogho?

File: lib/phx_tools_web/live/phx_tools_live/index.ex [hunk 20]
```diff
@@ -2,31 +2,55 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
...
 
-  defp apply_action(socket, :linux) do
-    assign(socket, seo_attributes: %{url: url(~p"/linux")})
+  defp installation_command(live_action) do
+    "/bin/bash -c \"$(curl -fsSL #{Endpoint.url() <> "/#{live_action}.sh"})\""
   end
 
-  defp apply_action(socket, :macOS) do
-    assign(socket, seo_attributes: %{url: url(~p"/macOS")})
+  defp decapitalize_os_name_first_letter(<<first::utf8, rest::binary>>) do
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_live/index.ex:51]: Why not use "Linux" and "macOS" everywhere to avoid doing this kind of stuff? I mean, using capital "L" and lowercase "m".

File: coveralls.json [hunk 21]
```diff
@@ -1,6 +1,6 @@
 {
   "coverage_options": {
-    "minimum_coverage": 100.0
+    "minimum_coverage": 99.0
```
Comment (almirsarajcic) [inline coveralls.json:3]: Can you check the HTML Coveralls generates to see which lines were not covered?
Run `mix ecoveralls.html && open cover/excoveralls.html`.

File: lib/phx_tools_web/live/live_session_helper.ex [hunk 22]
```diff
@@ -13,7 +13,7 @@ defmodule PhxToolsWeb.LiveSessionHelper do
         %{"operating_system" => "Linux"}
 
       "Mac" ->
-        %{"operating_system" => "Mac"}
+        %{"operating_system" => "MacOS"}
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/live_session_helper.ex:16]:
```suggestion
        %{"operating_system" => "macOS"}
```

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 23]
```diff
@@ -0,0 +1,275 @@
...
+              }
+              id={"asciinema-#{Utils.random_id()}"}
+              phx-hook="AsciinemaHook"
+            >
+            </div>
+
+            <p class="md:my-5 text-white md:px-12 text-xs md:text-sm lg:text-base font-martian">
+              Read about website updates here -
+              <a
+                href="https://optimum.ba/blog/phx-tools-complete-development-environment-for-elixir-and-phoenix"
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_components.ex:164]: The newest blog post will reside at https://optimum.ba/blog/exciting-updates-to-phx-tools.

File: test/phx_tools_web/live/index_test.exs [hunk 24]
```diff
@@ -3,44 +3,58 @@ defmodule PhxToolsWeb.PhxToolsLive.IndexTest do
...
-      {:ok, landing_live, _html} = live(conn, "/")
+      {:ok, _landing_live, html} = live(conn, "/")
 
-      refute has_element?(landing_live, "#confirmation")
+      assert html =~ "Unsupported Operating System Detected"
     end
   end
 
-  describe "Linux instructions page" do
+  describe "/linux" do
```
Comment (almirsarajcic) [inline test/phx_tools_web/live/index_test.exs:57]:
```suggestion
  describe "/Linux" do
```
Here and everywhere else?

**Review by almirsarajcic — CHANGES_REQUESTED** [review]

File: assets/js/hooks/copy_hook.js (hunk 19 above)
Comment (rowah) [reply assets/js/hooks/copy_hook.js:19]: Yes, she suggested it was taking a bit long.
There is a little difference between 800 and 1000,
 how about 600 milisecs?

File: lib/phx_tools_web/live/phx_tools_live/index.ex (hunk 20 above)
Comment (rowah) [reply lib/phx_tools_web/live/phx_tools_live/index.ex:51]: Because I am assigning the source code url dynamically like so
"https://github.com/optimumBA/phx.tools/blob/main/priv/static/#{action}.sh"
I could not use operating system because that does not change.

File: assets/js/hooks/copy_hook.js (hunk 19 above)
Comment (almirsarajcic) [reply assets/js/hooks/copy_hook.js]: Sorry, I didn't check. Just wanted to make sure Oghogho was ok with 400. I guess no changes are needed.

File: coveralls.json [hunk 25]
```diff
@@ -1,6 +1,6 @@
 {
   "coverage_options": {
-    "minimum_coverage": 100.0
+    "minimum_coverage": 100
```
Comment (almirsarajcic) [inline coveralls.json:3]: Can't remember why, but I started using floats instead of integers after seeing Coveralls failing unexpectedly.
```suggestion
    "minimum_coverage": 100.0
```

File: lib/phx_tools_web/live/phx_tools_live/index.ex [hunk 26]
```diff
@@ -2,31 +2,49 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
...
   end
 
-  defp assign_os(socket, %{"operating_system" => operating_system}),
-    do: assign(socket, :operating_system, operating_system)
+  defp assign_os_and_source_code_url(socket, %{"operating_system" => operating_system}) do
+    socket
+    |> assign(:operating_system, operating_system)
+    |> assign(
+      :source_code_url,
+      "https://github.com/optimumBA/phx.tools/blob/main/priv/static/#{operating_system}.sh"
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_live/index.ex:20]: I see this URL in two places.

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 27]
```diff
@@ -0,0 +1,268 @@
...
+          <Icons.os_icon os_name={@os_name} />
+        </div>
+        <h1 class="text-center text-white text-sm md:text-base font-martian"><%= @os_name %></h1>
+      </div>
+    </.link>
+    """
+  end
+
+  defp render_instructions(live_action) do
+    case live_action do
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_components.ex:231]: We can do the pattern matching in the function head, i.e. `defp render_instructions(:Linux) do`

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 28]
```diff
@@ -0,0 +1,268 @@
...
+      class="hidden font-normal solved-height font-martian bg-[#110A33] absolute md:w-[90%]"
+    >
+      <div class="block">
+        <div class="h-full shadow-[#C2B8FF] shadow-md rounded-md pb-2">
+          <div class="text-start px-[3%] lg:text-xl md:text-lg sm:text-md">
+            <h1 class="text-white text-center text-sm md:text-base lg:text-xl lg:my-[5%] md:my-[2%] sm:my-[2%] lg:pt-5">
+              <%= @live_action %> installation process
+            </h1>
+            <ol class="list-decimal ml-3 pl-5 text-xs md:text-sm lg:text-base text-white lg:mt-4 sm:mt-2 leading-6">
+              <%= for instruction <- render_instructions(@live_action) do %>
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_components.ex:194]: I feel like this could be better with slots.
https://hexdocs.pm/phoenix_live_view/Phoenix.Component.html#module-slot-attributes

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 29]
```diff
@@ -0,0 +1,268 @@
...
+
+  @spec command_select_button(assigns()) :: rendered()
+  def command_select_button(assigns) do
+    ~H"""
+    <.os_link_button
+      id={if @live_action == :macOS, do: "Linux", else: "macOS"}
+      href={if @live_action == :macOS, do: ~p"/Linux", else: ~p"/macOS"}
+      os_name={if @live_action == :Linux, do: "macOS", else: "Linux"}
+      class="px-3 flex justify-center"
+    />
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_components.ex:20]: Seems like this could be done inside the component with `Atom.to_string(@live_action)` or "#{@live_action}". Maybe pass the os attribute instead:
```suggestion
    <.os_link_button
      class="px-3 flex justify-center"
      os={Atom.to_string(@live_action)}
    />
```
and then use `@os` inside the component.

**Review by almirsarajcic — CHANGES_REQUESTED** [review]

File: lib/phx_tools_web/live/phx_tools_components.ex (hunk 29 above)
Comment (rowah) [reply lib/phx_tools_web/live/phx_tools_components.ex:20]: I just realised the function calling this may actually not be necessary anymore since I can use this directly.

File: lib/phx_tools_web/live/phx_tools_live/index.ex (hunk 26 above)
Comment (rowah) [reply lib/phx_tools_web/live/phx_tools_live/index.ex:20]: I decided to assign it dynamically on mount using the detected OS, but every time an action is applied I have to update it using the applied action.

File: assets/js/hooks/copy_hook.js (hunk 19 above)
Comment (rowah) [reply assets/js/hooks/copy_hook.js]: She's okay with 600

File: lib/phx_tools_web/live/phx_tools_live/index.ex (hunk 26 above)
Comment (almirsarajcic) [reply lib/phx_tools_web/live/phx_tools_live/index.ex:20]: Can't you have this in some private function that's called from both places or something?

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 30]
```diff
@@ -0,0 +1,273 @@
...
+  end
+
+  attr :class, :string
+  attr :live_action, :atom, required: true
+  attr :operating_system, :string, required: true
+
+  @spec command_select_button(assigns()) :: rendered()
+  def command_select_button(assigns) do
+    ~H"""
+    <.link href={if Atom.to_string(@live_action) == "macOS", do: ~p"/linux", else: ~p"/macOS"}>
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_components.ex:218]: Again, a lot of ifs in this function could be just pattern matching inside some smaller function components.

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 31]
```diff
@@ -0,0 +1,273 @@
...
+    </div>
+    """
+  end
+
+  attr :class, :string
+  attr :live_action, :atom, required: true
+  attr :operating_system, :string, required: true
+
+  @spec command_select_button(assigns()) :: rendered()
+  def command_select_button(assigns) do
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_components.ex:216]: Here's a hack for you: if you make the function component private, you don't need to write specs ;)
Maybe not even the attrs above.

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 32]
```diff
@@ -0,0 +1,273 @@
...
+        </div>
+        <h1 class="text-center text-white text-sm md:text-base font-martian">
+          <%= if @live_action == :linux, do: "macOS", else: "Linux" %>
+        </h1>
+      </div>
+    </.link>
+    """
+  end
+
+  defp render_instructions(:linux) do
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_components.ex:237]: The name is misleading now that this only returns a list of instructions.

File: lib/phx_tools_web/live/phx_tools_live/icons.ex [hunk 33]
```diff
@@ -0,0 +1,426 @@
...
+          <stop offset="0.5" stop-color="#20BDFF" />
+          <stop offset="1" stop-color="#A5FECB" />
+        </linearGradient>
+      </defs>
+    </svg>
+    """
+  end
+
+  @spec os_icon(assigns()) :: rendered()
+  def os_icon(%{os_to_switch_to: "macOS"} = assigns) do
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_live/icons.ex:106]:
```suggestion
  def os_icon(%{os: "macOS"} = assigns) do
```
?

File: test/phx_tools_web/live/index_test.exs [hunk 34]
```diff
@@ -51,12 +65,11 @@ defmodule PhxToolsWeb.PhxToolsLive.IndexTest do
     end
   end
 
-  describe "macOS instructions page" do
+  describe "/macOS" do
```
Comment (almirsarajcic) [inline test/phx_tools_web/live/index_test.exs:68]: Does mixing cases work (/macOS vs /macos)? We've had an issue with /Linux in the browsers. What about /macOS?

**Review by almirsarajcic — CHANGES_REQUESTED** [review]
There's an extra file `priv/static/Linux.sh`. Please remove.

File: test/phx_tools_web/live/index_test.exs (hunk 34 above)
Comment (rowah) [reply test/phx_tools_web/live/index_test.exs:68]: About this, I think it is good for readability, plus it's been in place and has not caused any issues.
It only came up when we switched to /Linux
I also made this change you are suggesting already, I don't know why it's not reflecting here.

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 35]
```diff
@@ -0,0 +1,273 @@
...
+        </div>
+        <p class="text-center text-white font-martian text-xs md:text-sm lg:text-base leading-5 md:leading-6 sm:py-8">
+          Phx.tools is a shell script for Linux and macOS that configures the development environment for you in a few easy steps. Once you finish running the script, you'll be able to start the database server, create a new Phoenix application, and launch the server.
+        </p>
+      </div>
+      <div class="">
+        <p class="text-white font-martian text-center text-xs md:text-sm lg:text-base leading-6 sm:pb-4">
+          Read about website updates here -
+          <a
+            href="https://optimum.ba/blog/exciting-updates-to-phx-tools"
```
[the remark this answers is not in the document]
Comment (rowah) [reply lib/phx_tools_web/live/phx_tools_components.ex:77]: I saw it too small to be in a separate component.
I have separated though, and it looks better this way.

**Comment by rowah** [comment]
There is an empty Linux.sh file which I think is the one you see as the extra file, I somehow introduced it with one of my previous commits (https://github.com/optimumBA/phx.tools/pull/19/commits/6248f3b9360bb31e424d3191e6455654da5fe93a)
 but now although github shows its an empty file, deleting it actually deletes the whole Linux scripts.
I don't get what's going on.

**Comment by rowah** [comment]
> There's an extra file `priv/static/Linux.sh`. Please remove.
There is an empty Linux.sh file which is the one you see as the extra file, I somehow introduced it with one of my previous commits (https://github.com/optimumBA/phx.tools/pull/19/commits/6248f3b9360bb31e424d3191e6455654da5fe93a)
 but now although github shows its an empty file, deleting it actually deletes the whole Linux scripts.
I don't get what's going on.

File: lib/phx_tools_web/live/phx_tools_live/index.ex [hunk 36]
```diff
@@ -2,31 +2,56 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
...
-    assign(socket, seo_attributes: %{url: url(~p"/linux")})
+  defp installation_command(live_action) do
+    "/bin/bash -c \"$(curl -fsSL #{Endpoint.url() <> "/#{capitalize_os_name_first_letter("#{live_action}")}.sh"})\""
   end
 
-  defp apply_action(socket, :macOS) do
-    assign(socket, seo_attributes: %{url: url(~p"/macOS")})
+  defp source_code_url(os) do
+    "https://github.com/optimumBA/phx.tools/blob/main/priv/static/#{capitalize_os_name_first_letter("#{os}")}.sh"
   end
```
[the remark this answers is not in the document]
Comment (rowah) [reply lib/phx_tools_web/live/phx_tools_live/index.ex]: This may have been possible if I was not using the capitalize_os_name_first_letter (now get_os_from_live_action) elsewhere where the .sh extension is not needed.

**Comment by almirsarajcic** [comment]
> > There's an extra file `priv/static/Linux.sh`. Please remove.
>
> There is an empty Linux.sh file which is the one you see as the extra file, I somehow introduced it with one of my previous commits ([6248f3b](https://github.com/optimumBA/phx.tools/commit/6248f3b9360bb31e424d3191e6455654da5fe93a))
> but now although github shows its an empty file, deleting it actually deletes the whole Linux scripts.
> I don't get what's going on.

That's because of the case change.

File: lib/phx_tools_web/live/phx_tools_components.ex [hunk 37]
```diff
@@ -0,0 +1,288 @@
...
+  end
+
+  attr :installation_instructions, :list
+  attr :live_action, :atom
+  attr :operating_system, :string
+
+  slot :inner_block, required: true
+
+  @spec os_instructions(assigns()) :: rendered()
+  def os_instructions(assigns) do
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_components.ex:164]: If the function is not used outside of the module, you can make it private (and remove attrs and specs).

File: test/phx_tools_web/live/index_test.exs (hunk 34 above)
Comment (almirsarajcic) [inline test/phx_tools_web/live/index_test.exs:68]:
```suggestion
  describe "/macos" do
```

File: lib/phx_tools_web/live/phx_tools_live/index.ex [hunk 38]
```diff
@@ -2,31 +2,60 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
...
+  end
+
+  defp source_code_url(live_action) do
+    "https://github.com/optimumBA/phx.tools/blob/main/priv/static/#{get_os_from_live_action("#{live_action}")}.sh"
+  end
+
+  @spec get_os_from_live_action(String.t()) :: String.t()
+  def get_os_from_live_action("linux"), do: "Linux"
+
+  def get_os_from_live_action("macOS"), do: "macOS"
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_live/index.ex:58]:
```suggestion
  def get_os_from_live_action("macos"), do: "macOS"
```

File: lib/phx_tools_web/live/phx_tools_live/index.ex [hunk 39]
```diff
@@ -2,31 +2,60 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
...
+  def handle_params(_params, _uri, socket) do
+    apply_action(socket, socket.assigns.live_action)
+  end
 
-  defp apply_action(socket, :linux) do
-    assign(socket, seo_attributes: %{url: url(~p"/linux")})
+  defp apply_action(socket, :index) do
+    case socket.assigns.operating_system do
+      "Linux" -> {:noreply, assign(socket, :live_action, :linux)}
+      "macOS" -> {:noreply, assign(socket, :live_action, :macOS)}
```
Comment (almirsarajcic) [inline lib/phx_tools_web/live/phx_tools_live/index.ex:35]:
```suggestion
      "macOS" -> {:noreply, assign(socket, :live_action, :macos)}
```
Are there any other places where you need to make this change?

**Review by almirsarajcic — CHANGES_REQUESTED** [review]

File: lib/phx_tools_web/live/phx_tools_live/index.ex (hunk 39 above)
Comment (rowah) [reply lib/phx_tools_web/live/phx_tools_live/index.ex:35]: Checking and updating

**Comment by almirsarajcic** [comment]
Looks good. We just need to change the IDs of ASCIINEMA recordings after they are done and we can merge this.

**Review by almirsarajcic — APPROVED** [review]
